---
title: ""
output:
  html_document:
    toc: true
    toc_float: true
    collapsed: false
    number_sections: false
    toc_depth: 1
    #code_folding: hide
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(message=FALSE,warning=FALSE, cache=TRUE)
```

```{r include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(png)
library(imager)
library(tidyverse)
library(tximport)
library(DESeq2)
library(GenomicFeatures)
library(rtracklayer)
library(genefilter)
library(ggplot2)
library(grDevices)
library(RColorBrewer)
library(gplots)
library(ggrepel)
library(dplyr)
library(pheatmap)
library(plotly)
library(heatmaply)
library(tidyverse)
```

# PCA {.tabset}

We discussed aggregating the samples P16, pCDH-empty and K36WT as a potential control to compare treated samples against. However, it seems that the K36WT is more similar to K36R than P16 or pCDH-empty. The relationships are recapitulated in the heatmap below. We see slight differences in the PCAs and heatmaps between (varianceStabilizingTransformation()) and (rlogTransformation()).

Looking at the rlog PCA and heatmap, it seems appropriate to use P16 and pCDH-empty as a control and remove K36WT. 

Note: pCDH indicates they're derived from over-expression experiments using pCDH lentivectors.

<details>
<summary>Size Factors</summary>
<pre>
```{r echo=FALSE}
setwd("~/testGitHub/Testing")

load("dds.RData")
load("mdat.RData")
load("vsd.RData")
load("rld.RData")

size_factors <- data.frame(estimateSizeFactorsForMatrix(counts(dds)))
names(size_factors)[names(size_factors) == "estimateSizeFactorsForMatrix.counts.dds.."] <- "Size Factor"
size_factors
```
</pre>
</details>

## vst
```{r echo=FALSE}
rv <- rowVars(assay(vsd))
select <- order(rv, decreasing=T)[seq_len(min(500,length(rv)))]
pc <- prcomp(t(assay(vsd)[select,]))
condition <- mdat$cond
scores <- data.frame(pc$x, condition)
percentage <- round(pc$sdev / sum(pc$sdev) * 100, 2)
percentage <- paste( colnames(scores), "(", paste( as.character(percentage), "%", ")", sep="") )

p <- plot_ly(scores,x=scores$PC1,y=scores$PC2,text=rownames(scores),mode="markers",color=factor(condition),marker=list(size=11))
p <- layout(p,title="",   xaxis = list(title = percentage[1]),
                 yaxis = list(title = percentage[2]))
p

```

## rlog
```{r echo=FALSE}
rv <- rowVars(assay(rld))
select <- order(rv, decreasing=T)[seq_len(min(500,length(rv)))]
pc <- prcomp(t(assay(rld)[select,]))
condition <- mdat$cond
scores <- data.frame(pc$x, condition)
percentage <- round(pc$sdev / sum(pc$sdev) * 100, 2)
percentage <- paste( colnames(scores), "(", paste( as.character(percentage), "%", ")", sep="") )

p <- plot_ly(scores,x=scores$PC1,y=scores$PC2,text=rownames(scores),mode="markers",color=factor(condition),marker=list(size=11))
p <- layout(p,title="",   xaxis = list(title = percentage[1]),
                 yaxis = list(title = percentage[2]))
p

```

# Heatmap {.tabset}

## vst
```{r echo=FALSE}
sampleDists <- dist(t(assay(vsd[select])))
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(vsd$cond, vsd$dir, sep="-")
colnames(sampleDistMatrix) <- paste(vsd$cond, vsd$dir, sep="-")
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
heatmaply(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors)

```

## rlog
```{r echo=FALSE}
sampleDists <- dist(t(assay(rld[select])))
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(rld$cond, rld$dir, sep="-")
colnames(sampleDistMatrix) <- paste(rld$cond, rld$dir, sep="-")
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
heatmaply(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors)

```

# Overlapping Genes

It doesn't really make sense to do pathway analysis on the overlapping genes for the Detroit samples because there are so few genes. However, I did investigate the partial overlap with most DGE genes - between K36M and NSD1KO (111 upregulated genes). *Please see the fourth page (DAVID analyses) for the results.*

Criteria for DGE: 2-fold change in gene expression compared to control and p.adj < 0.05

<table>
  <tr>
    <td align="middle">**Upregulated Genes**</td>
     <td align="middle">**Downregulated Genes**</td>
  </tr>
  <tr>
    <td valign="top"><img src="C:/users/Reinnier Padilla/Documents/testGitHub/Testing/upregulated_venn_diagramm.png"></td>
    <td valign="top"><img src="C:/users/Reinnier Padilla/Documents/testGitHub/Testing/downregulated_venn_diagramm_Detroit.png"></td>
  </tr>
 </table>

